python/shotgun_fields/image_widget.py

```diff
@@ -279,24 +318,53 @@ def _replace_image(self):
         """
         self._upload_image()
 
-    def _scale_pixmap(self):
+    def _scale_pixmap(self, pixmap):
         """
         Scale the pixmap in preparation for display.
         """
-        scaled_pixmap = self._pixmap.scaled(
+        return pixmap.scaled(
             self.size(),
             QtCore.Qt.KeepAspectRatio,
             QtCore.Qt.SmoothTransformation
         )
-        self._scaled_width = scaled_pixmap.width()
-        super(ImageWidget, self).setPixmap(scaled_pixmap)
-        self._update_btn_position()
 
     def _show_image(self):
         """
-        The user requested to show the file. Forward to desktop services.
+        The user requested to show the file.
+
+        Display the image in a transient, modeless QDialog.
         """
-        QtGui.QDesktopServices.openUrl("file:///%s" % (self._image_url))
+
+        # don't continue unless there's somethign to show
+        if not self._pixmap:
+            return
+
+        if self._image_url:
+            # try to use the path if there is one
+            display_pixmap = QtGui.QPixmap(self._image_url)
+        else:
+            # may only have the pixmap (delegates)
+            display_pixmap = QtGui.QPixmap(self._pixmap)
+
+        # construct the dialog, display label, and a button box to close it.
+
+        img_dialog = QtGui.QDialog(self)
+        img_dialog.setWindowTitle(
+            "Viewing: %s %s" % (self._entity_type, self._field_name)
+        )
+
+        lbl = QtGui.QLabel()
```

Comment:
The popup dialog is just the label with an `Ok` button. My assumption was that this would always be a relatively small image, but maybe that's wrong. Does SG limit/scale the `image` field uploads on the backend? What do you think is the right way to handle this? I'm not tied to this at all. :)